doc/design/speech/README.MD

```diff
@@ -140,7 +140,17 @@ TODO by Assignees
 
 ### Beam Search with CTC and LM
 
-TODO by Assignees
+<div align="center">
+<img src="image/beam_search.png" width=400><br/>
+Figure 2. Algorithm for Beam Search Decoder.
+</div>
+
+- The **Beam Search Decoder** for DS2 CTC-trained network follows the similar approach in \[[3](#references)\] with a modification for the ambiguous part, as shown in Figure 2.
+- An **external defined scorer** would be passed into the decoder to evaluate a candidate prefix during decoding whenever a space character appended.
+- Such scorer is a unified class, may consisting of language model, word count or any customed evaluators.
```

Comment:
1. --> any other custom scorers.
2. What do you mean by "a unified class"?
3. Remove "may".

Reply:
Done and modified.